Source/HTTP/Android/android_http_request.cpp

```diff
@@ -1,9 +1,16 @@
 #include "pch.h"
 
 #include "android_http_request.h"
+#include "http_android.h"
 #include <httpClient/httpClient.h>
 #include <vector>
 
+// The following symbols live in files that are not referenced by any other c
+// file and are only called by JNI callers. In order to force the compiler to
+// not ignore the file we make a token dummy reference to the file here.
+volatile auto dummyHCRequestOnRequestCompleted = &Java_com_xbox_httpclient_HttpClientRequest_OnRequestCompleted;
```

Comment:
why only do symbols for two of them? woulda thought we'd do one or all

Reply:
That's what I thought too, but the other two functions are linking just fine, so I didn't bother

Reply:
do we need both referenced here for both to link properly?

Reply:
yes, `OnRequestCompleted` and `OnRequestFailed` need to be referenced otherwise the link won't happen